src/server_manager/web_app/cloud_accounts.ts

```diff
@@ -12,41 +12,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-export interface TokenManager {
-  // Returns the Oauth token, or null if unavailable.
-  getStoredToken(): string;
-  // Writes the token to storage.
-  writeTokenToStorage(token: string): void;
-  // Removes the token from storage.
-  removeTokenFromStorage(): void;
-}
+import {DigitalOceanSession} from '../cloud/digitalocean_api';
+import {DigitalOceanAccount} from './digitalocean_account';
+
+type DigitalOceanSessionFactory = (accessToken: string) => DigitalOceanSession;
+type DigitalOceanAccountFactory = (session: DigitalOceanSession) => DigitalOceanAccount;
 
 // TODO: this class combines URL manipulation with persistence logic.
 // Consider moving the URL manipulation logic to a separate class, so we
 // can pass in other implementations when the global "window" is not present.
-export class DigitalOceanTokenManager implements TokenManager {
+export class CloudAccounts {
   private readonly DIGITALOCEAN_TOKEN_STORAGE_KEY = 'LastDOToken';
 
-  // Searches the current URL (post-OAuth) and local storage for a DigitalOcean
-  // access token. The token is not checked for validity as this would require
-  // an extra roundtrip to DigitalOcean.
-  getStoredToken(): string {
-    const tokenFromStorage = this.getTokenFromStorage();
-    if (tokenFromStorage) {
-      console.info('found access token in local storage');
-      return tokenFromStorage;
-    }
+  constructor(
+      private digitalOceanSessionFactory: DigitalOceanSessionFactory,
+      private digitalOceanAccountFactory: DigitalOceanAccountFactory) {}
 
-    // Not an error as user may not yet have authenticated.
-    return null;
+  connectDigitalOceanAccount(token: string): DigitalOceanAccount {
+    this.writeTokenToStorage(token);
+    return this.getDigitalOceanAccount();
   }
 
-  writeTokenToStorage(token: string): void {
-    localStorage.setItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY, token);
+  disconnectDigitalOceanAccount(): void {
+    localStorage.removeItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY);
   }
 
-  removeTokenFromStorage(): void {
-    localStorage.removeItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY);
+  getDigitalOceanAccount(): DigitalOceanAccount {
+    const token = this.getTokenFromStorage();
+    if (token) {
+      const digitalOceanSession = this.digitalOceanSessionFactory(token);
```

Comment:
We can probably fold both into one digitalOceanAccountFactory(token)
The caller never uses the session.